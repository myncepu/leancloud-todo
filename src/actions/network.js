export const CHANGE_CONNECTION_STATUS = 'CHANGE_CONNECTION_STATUS'

export const changeConectionStatus = connectionInfo => ({
  type: CHANGE_CONNECTION_STATUS,
  connectionInfo,
})
