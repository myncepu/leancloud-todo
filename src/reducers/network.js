import { CHANGE_CONNECTION_STATUS } from '../actions/network'

const isConnected = (status) => {
  if (status.toLowerCase() === 'none') {
    return false
  }
  return true
}

const initialState = {
  connectionInfo: null,
  hasCheckedStatus: false,
  isConnected: false,
}

const networkReducer = (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_CONNECTION_STATUS:
      return {
        ...state,
        connectionInfo: action.connectionInfo,
        isConnected: isConnected(action.connectionInfo.type),
        hasCheckedStatus: true,
      }
    default:
      return state
  }
}

export default networkReducer
