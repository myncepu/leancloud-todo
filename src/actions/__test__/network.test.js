import { changeConectionStatus } from '../network'

describe('changeConectionStatus', () => {
  it('creates a properly formatted action', () => {
    expect(changeConectionStatus()).toMatchSnapshot()
  })
})
