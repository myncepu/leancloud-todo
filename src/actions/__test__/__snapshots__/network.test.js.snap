// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeConectionStatus creates a properly formatted action 1`] = `
Object {
  "connectionInfo": undefined,
  "type": "CHANGE_CONNECTION_STATUS",
}
`;
